projects/laji/src/app/shared-modules/observation-map/observation-map/observation-visualization.ts

```diff
@@ -0,0 +1,299 @@
+      {
+        color: '#90dacf',
+        //label: 'Community verified'
+        translate: 'laji-map.legend.mode.recordQuality.2'
+      },
+      {
+        color: '#ffffbf',
+        //label: 'Neutral'
+        translate: 'laji-map.legend.mode.recordQuality.3'
+      },
+      {
+        color: '#fdbf66',
+        //label: 'Uncertain'
+        translate: 'laji-map.legend.mode.recordQuality.4'
+      },
+      {
+        color: '#f26840',
+        //label: 'Erroneous'
+        translate: 'laji-map.legend.mode.recordQuality.5'
+      }
+    ],
+    getFeatureStyle: (options) => ({
+      ...baseFeatureStyle,
+      color: getRecordQualityColor(options.feature.properties.recordQualityMax),
+      className: getCoordinateAccuracyClassName(options.feature.properties['gathering.interpretations.coordinateAccuracy'])
+    })
+  },
+  redlistStatus: {
+    label: 'laji-map.legend.mode.redlistStatus',
+    categories: [
+      {
+        color: '#348cf0',
+        label: 'LC, DD, NA, NE'
+      },
+      {
+        color: '#90dacf',
+        label: 'NT'
+      },
+      {
+        color: '#ffffbf',
+        label: 'VU'
+      },
+      {
+        color: '#fdbf66',
+        label: 'EN'
+      },
+      {
+        color: '#f26840',
+        label: 'CR, RE, EW, EX'
+      }
+    ],
+    getFeatureStyle: (options) => ({
+      ...baseFeatureStyle,
+      color: getRedlistStatusColor(options.feature.properties.redListStatusMax),
+      className: getCoordinateAccuracyClassName(options.feature.properties['gathering.interpretations.coordinateAccuracy'])
+    })
+  },
+  individualCount: {
+    label: 'laji-map.legend.mode.individualCount',
+    categories: [
+      {
+        color: '#ffffff',
+        label: '0'
+      },
+      {
+        color: '#348cf0',
+        label: '1-5'
+      },
+      {
+        color: '#90dacf',
+        label: '5-20'
+      },
+      {
+        color: '#ffffbf',
+        label: '20-50'
+      },
+      {
+        color: '#fdbf66',
+        label: '50-100'
+      },
+      {
+        color: '#f26840',
+        label: '> 100'
+      }
+    ],
+    getFeatureStyle: (options) => ({
+      ...baseFeatureStyle,
+      color: getIndividualCountColor(options.feature.properties.individualCountSum),
+      className: getCoordinateAccuracyClassName(options.feature.properties['gathering.interpretations.coordinateAccuracy'])
+    })
+  },
+  recordAge: {
+    label: 'laji-map.legend.mode.recordAge',
+    categories: [
+      {
+        color: '#348cf0',
+        label: `≥ ${currentYear - yearsAgoBreakpoints[0]}`
+      },
+      {
+        color: '#90dacf',
+        label: `≥ ${currentYear - yearsAgoBreakpoints[1]}`
+      },
+      {
+        color: '#ffffbf',
+        label: `≥ ${currentYear - yearsAgoBreakpoints[2]}`
+      },
+      {
+        color: '#fdbf66',
+        label: `≥ ${currentYear - yearsAgoBreakpoints[3]}`
+      },
+      {
+        color: '#f26840',
+        label: `< ${currentYear - yearsAgoBreakpoints[3]}`
+      }
```

Comment:
Is the `yearsAgoBreakpoints[3]` twice on purpose here? This is why hard coded indices can produce bugs easily (first doing copy pasting and manually hard coding the changes is prone to typos, then after doing that a several times your eyes just see a repetitive pattern and don't notice that one of the indices is wrong. It's like trying to find Waldo). This could be more readable if for example there was a single data structure that would hold the year breakpoints and the colors.

Reply:
`yearsAgoBreakpoints[3]` is twice on purpose.

Reply:
I made this a little clearer by moving the `currentYear - breakpoint` calculation to the `RECORD_AGE_BREAKPOINTS` array, but I'm not sure if there's anything else that can be done. The values can't be written directly into the string/enum even if we wanted to since it depends on current year (which is dynamic). Besides honestly I think the logic here is pretty clear?

Reply:
I was wrong about the label, it's supposed to be 
`< 1982` like it is now. Sorry :D I'll mark this as resolved